### PR TITLE
Add git package to Mariner images

### PR DESCRIPTION
**Pull request by dagood**
* Another package for AzDO suitability. For https://github.com/microsoft/go/issues/1164

### CODE REVIEW

File: patches/0002-Add-Mariner-FIPS-package-upgrade.patch [hunk 1]
```diff
@@ -105,8 +105,9 @@ index 1c8c549b281edc..6e3544787124cc 100644
 +		# To reduce image size, CBL-Mariner image doesn't include full ca-certificates: https://github.com/microsoft/CBL-Mariner/issues/3563
```
Comment (karianna) [inline patches/0002-Add-Mariner-FIPS-package-upgrade.patch]:
```suggestion
+		# To reduce image size, the CBL-Mariner (aka Azure Linux) image doesn't include full ca-certificates: https://github.com/microsoft/CBL-Mariner/issues/3563
```

File: patches/0002-Add-Mariner-FIPS-package-upgrade.patch [hunk 2]
```diff
@@ -105,8 +105,9 @@ index 1c8c549b281edc..6e3544787124cc 100644
...
 +		# Full certs are expected in a builder-style image (e.g. to fetch dependencies), so install them.
```
Comment (karianna) [inline patches/0002-Add-Mariner-FIPS-package-upgrade.patch]:
```suggestion
+		# Full certs are expected in a builder-style image (e.g., to fetch dependencies), so we need to install them.
```

File: patches/0002-Add-Mariner-FIPS-package-upgrade.patch [hunk 3]
```diff
@@ -105,8 +105,9 @@ index 1c8c549b281edc..6e3544787124cc 100644
...
 +		ca-certificates \
-+		# Azure Pipelines expects "useradd", "groupadd", and "su".
++		# Azure Pipelines expects various tools that would be in a buildpack-deps based image.
```
Comment (karianna) [inline patches/0002-Add-Mariner-FIPS-package-upgrade.patch]: Can we add a link to buildpack-deps docs?

File: patches/0002-Add-Mariner-FIPS-package-upgrade.patch [hunk 4]
```diff
@@ -105,8 +105,9 @@ index 1c8c549b281edc..6e3544787124cc 100644
...
 +		# These are included by default in Debian but not CBL-Mariner. https://github.com/microsoft/azurelinux/issues/8964
```
Comment (karianna) [inline patches/0002-Add-Mariner-FIPS-package-upgrade.patch]: Can we open an ADO work item to review this next semester.

File: src/microsoft/1.21/cbl-mariner2.0/Dockerfile [hunk 5]
```diff
@@ -100,8 +100,9 @@ RUN set -eux; \
 		# To reduce image size, CBL-Mariner image doesn't include full ca-certificates: https://github.com/microsoft/CBL-Mariner/issues/3563
 		# Full certs are expected in a builder-style image (e.g. to fetch dependencies), so install them.
 		ca-certificates \
-		# Azure Pipelines expects "useradd", "groupadd", and "su".
+		# Azure Pipelines expects various tools that would be in a buildpack-deps based image.
```
Comment (karianna) [inline src/microsoft/1.21/cbl-mariner2.0/Dockerfile]: Same comment about linking as above

File: src/microsoft/1.22/cbl-mariner2.0/Dockerfile [hunk 6]
```diff
@@ -100,8 +100,9 @@ RUN set -eux; \
 		# To reduce image size, CBL-Mariner image doesn't include full ca-certificates: https://github.com/microsoft/CBL-Mariner/issues/3563
 		# Full certs are expected in a builder-style image (e.g. to fetch dependencies), so install them.
 		ca-certificates \
-		# Azure Pipelines expects "useradd", "groupadd", and "su".
+		# Azure Pipelines expects various tools that would be in a buildpack-deps based image.
```
Comment (karianna) [inline src/microsoft/1.22/cbl-mariner2.0/Dockerfile]: Same comment about linking as above

**Review by karianna — CHANGES_REQUESTED** [review]

File: patches/0002-Add-Mariner-FIPS-package-upgrade.patch (hunk 3 above)
Comment (dagood) [reply patches/0002-Add-Mariner-FIPS-package-upgrade.patch]: Done, I reorganized this part in general too to try to make it more direct. (And included a couple line breaks for readability.)

File: patches/0002-Add-Mariner-FIPS-package-upgrade.patch (hunk 4 above)
Comment (dagood) [reply patches/0002-Add-Mariner-FIPS-package-upgrade.patch]: Opened https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2055698 and added a link.

File: src/microsoft/1.21/cbl-mariner2.0/Dockerfile (hunk 5 above)
Comment (dagood) [reply src/microsoft/1.21/cbl-mariner2.0/Dockerfile]: For future reference: the `Dockerfile`s are generated, and the CI (`microsoft-go-images (Repeatability Test)`) checks that the Dockerfiles match what the templates evaluate to. So, it usually isn't necessary to check these files other than to make sure the template's logic did the right thing.

(If you noticed that the GitHub review UI hides these files by default, this is driven by https://github.com/microsoft/go-images/blob/microsoft/main/.gitattributes, which we inherited from upstream--along with the template system.)

**Review by karianna — APPROVED** [review]

**Review by gdams — COMMENTED** [review]
Retrospective +1 from me

**Pull request merged**